origin-js/src/ipfsInterface/schemas/listing.json

```diff
@@ -78,8 +73,7 @@
       "required": [
         "startDate",
         "endDate",
-        "isAvailable",
-        "price"
```

Comment:
Good question... It's because the calendar has the concept of "unavailable" time slots, and a price isn't required in that case. Unavailable slots are used for overriding a repeating timeslot that is normally available, or for blocking out time on a calendar.
This could change in the future but for now, we can't make price required unless we force the dapp to put a price of "0" on all unavailable timeslots, which I'd be okay with, but this seemed cleaner IMO

Reply:
I see. Makes sense for now. In future I could see that we define two different type of timeslots: one for "blocked time" and the other that handles available slots. It is not that difficult to do with JSON schema, using the "oneOf" field. See for ex: https://github.com/OriginProtocol/origin/blob/7de3e3f336b2d12a1a58314ea0b303e2133a5559/origin-js/src/ipfsInterface/schemas/listing.json#L162